include/ensmallen_bits/moead/decomposition_policies/pbi_decomposition.hpp

```diff
@@ -63,11 +63,11 @@ class PenaltyBoundaryIntersection
   {
     typedef typename VecType::elem_type ElemType;
     //! A unit vector in the same direction as the provided weight vector.
-    const VecType referenceDirection = weight / arma::norm(weight, 1);
+    const VecType referenceDirection = weight / arma::norm(weight, 2);
     //! Distance of F(x) from the idealPoint along the reference direction.
     const ElemType d1 = arma::dot(candidateFitness - idealPoint, referenceDirection);
     //! The perpendicular distance of F(x) from reference direction.
-    const ElemType d2 = arma::norm(candidateFitness - (idealPoint + d1 * referenceDirection), 1);
+    const ElemType d2 = arma::norm(candidateFitness - (idealPoint + d1 * referenceDirection), 2);
```

Comment:
This change is related to @coatless [comment](https://github.com/mlpack/ensmallen/pull/296#discussion_r654672252) on L2 norm. I misunderstood that concept which reflected here.

The paper takes "distance" (i.e. L2 norm) for finding d1 and d2. Here's a reference pymoo implementation

https://github.com/msu-coinlab/pymoo/blob/b91212185ab04d006741372a4a480608b8d32e68/pymoo/decomposition/util.py#L5-L9

Reply:
@jonpsy should we make a utility function that wraps `norm(x, 2)` called `norm_l2()` to clarify intent and avoid future headaches?

Reply:
In armadillo? I think that'd be @conradsnicta's call

Reply:
No need to go upstream, this could be placed inside `ensmallen`.

By default, [`norm(x)` has `p = 2`](http://arma.sourceforge.net/docs.html#norm) per docs.

Reply:
I think mathematical abstractions should limit to armadillo. I'll change norm(x, 2) to norm then. That'll be better.

Reply:
@jonpsy that wasn't my intent by stating that! In fact, I'd prefer it with the explicit `norm(x, 2)` vs. `norm()`. It's almost always better to be explicit and avoid relying on defaults as they may inadvertently change. 

I'm of the mindset that a general function is great, but creating helper functions to commonly used tasks is ideal.

Reply:
@coatless I see, I guess I misinterpreted it. I think using ```arma::norm()``` is better. The general notion is that norm refers to  L2 norm unless otherwise stated. It looks like other libraries follow similar culture as well (see ref. implementation).